specification/trafficmanager/resource-manager/readme.md

```diff
@@ -51,6 +51,10 @@ directive:
     reason: This will cause breaking changes in .NET SDK
     from: trafficmanager.json
     where: $.definitions.Endpoint
+  - suppress: RequiredPropertiesMissingInResourceModel
+    reason: This will cause breaking changes in .NET SDK
+    from: trafficmanager.json
+    where: $.definitions.UserMetricsModel

```

Comment:
@tatlicioglu - this is to suppress the following ARMViolation I guess:

```
RequiredPropertiesMissingInResourceModel/R2020/ARMViolation: Model definition 'UserMetricsModel' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
```

Looking at the definition of UserMetricsModel it does have `name, id and type` in the hierarchy (UserMetricsModel -> ProxyResource -> [Resource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json)), but `id` is not readonly. 

As mentioned in the suppression reason, adding readonly attribute for Resource::id will cause breaking change in the SDK (e.g. [ctr](https://github.com/Azure/azure-sdk-for-net/blob/682e9583d4a69a275fcbab76278807a80b20418d/src/SDKs/TrafficManager/Management.TrafficManager/Generated/Models/UserMetricsModel.cs#L42) for UserMetricsModel will break) that requires a major version bump for the SDK. A major version upgrade of SDK for such a small change doesn't worth.

So we might want add the work of `marking id as readonly` as a TODO in major version upgrade work items. At this point i'm not sure when that happens. Maybe we should introduce a process of cleaning the swaggers once in a semester or so where we fix all these issues and do a major version upgrade.  Something that needs to be discussed within the team.

It looks like there are bunch of other models (Endpoint, Profile, TrafficManagerGeographicHierarchy & HeatMapModel) inherits from Resource has `RequiredPropertiesMissingInResourceModel`. Any reason not to suppress violations for those models?

\\cc @johanste, @salameer